webapp/packages/plugin-sql-editor-navigation-tab-script/src/PluginBootstrap.ts

```diff
@@ -201,6 +107,115 @@ export class PluginBootstrap extends Bootstrap {
       },
       getItems: (context, items) => [...items, ACTION_SAVE_AS_SCRIPT],
     });
+
+    this.keyBindingService.addKeyBindingHandler({
+      id: 'save-as-script',
+      binding: KEY_BINDING_SQL_EDITOR_SAVE_AS_SCRIPT,
+      actions: [ACTION_SAVE_AS_SCRIPT],
+      contexts: [DATA_CONTEXT_SQL_EDITOR_STATE],
+      isBindingApplicable: (_, action) => action === ACTION_SAVE_AS_SCRIPT,
+      handler: this.saveAsScriptHandler.bind(this),
+    });
+  }
+
+  private async saveAsScriptHandler(context: any, action: any) {
```

Comment:
Please add types to context arg. Also no need to check and pass action type as we already know it from a method name. Check it only in handler above